packages/flutter/lib/src/widgets/modal_button_sheet.dart

```diff
@@ -0,0 +1,652 @@
+        final velocity = _velocityTracker!.getVelocity().pixelsPerSecond.dy;
+        _velocityTracker = null;
+        _startTime = null;
+        _handleDragEnd(velocity);
+      }
+    }
+  }
+
+  Curve get _defaultCurve => widget.animationCurve ?? _modalBottomSheetCurve;
+
+  @override
+  void initState() {
+    animationCurve = _defaultCurve;
+    _bounceDragController =
+        AnimationController(vsync: this, duration: Duration(milliseconds: 300));
+
+    // Todo: Check if we can remove scroll Controller
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final bounceAnimation = CurvedAnimation(
+      parent: _bounceDragController,
+      curve: Curves.easeOutSine,
+    );
+
+    var child = widget.child;
+    if (widget.containerBuilder != null) {
+      child = widget.containerBuilder(
+        context,
+        widget.animationController,
+        child,
+      );
+    }
+
+    final accessibleNavigation =
+        MediaQuery.of(context)?.accessibleNavigation ?? false;
+
+    child = AnimatedBuilder(
+      animation: widget.animationController,
+      child: RepaintBoundary(child: child),
+      builder: (context, Widget? child) {
+        assert(child != null);
+        final animationValue = animationCurve.transform(
+            accessibleNavigation ? 1.0 : widget.animationController.value);
+
+        final draggableChild = !widget.enableDrag
+            ? child
+            : KeyedSubtree(
+                key: _childKey,
+                child: AnimatedBuilder(
+                  animation: bounceAnimation,
+                  builder: (context, _) => CustomSingleChildLayout(
+                    delegate: _CustomBottomSheetLayout(bounceAnimation.value),
+                    child: GestureDetector(
+                      onVerticalDragUpdate: (details) {
+                        _handleDragUpdate(details.delta.dy);
+                      },
+                      onVerticalDragEnd: (details) {
+                        _handleDragEnd(details.primaryVelocity);
+                      },
+                      child: NotificationListener<ScrollNotification>(
+                        onNotification: (ScrollNotification notification) {
+                          _handleScrollUpdate(notification);
+                          return false;
+                        },
+                        child: child!,
+                      ),
+                    ),
+                  ),
+                ),
+              );
+        return ClipRect(
+          child: CustomSingleChildLayout(
+            delegate: _ModalBottomSheetLayout(
+              animationValue,
+              widget.expanded,
+            ),
+            child: draggableChild,
+          ),
+        );
+      },
+    );
+
+    return ScrollToTopStatusBarHandler(
+      child: child,
+      scrollController: _scrollController,
+    );
+  }
+}
+
+class _ModalBottomSheetLayout extends SingleChildLayoutDelegate {
```

Comment:
I might be wrong but given the iOS animation, I'm somewhat tempted to think that we might not need to use something this involved.